### PR TITLE
Adds missing case to assigns clause type checking

### PR DESCRIPTION
**Pull request by feliperodri**
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by kroening** [comment]
BTW, `convert_type` is likely the wrong place to do the check. I'd move it to the typechecker.

File: src/solvers/flattening/boolbv.h [hunk 1]
```diff
@@ -79,7 +79,7 @@ class boolbvt:public arrayst
     SUB::post_process();
   }
 
-  enum class unbounded_arrayt { U_NONE, U_ALL, U_AUTO };
+  enum class unbounded_arrayt { U_AUTO, U_ALL, U_AUTO };
```
Comment (martin-cs) [inline src/solvers/flattening/boolbv.h]: This does not seem to be part of this PR.  Please have a look at https://github.com/diffblue/cbmc/pull/6232 .

**Review by martin-cs — CHANGES_REQUESTED** [review]
The changes to convert_type look good but I think the array changes should be handled separately.

File: src/solvers/flattening/boolbv.h (hunk 1 above)
Comment (feliperodri) [reply src/solvers/flattening/boolbv.h]: Sorry @martin-cs I send it here by mistake.

**Comment by feliperodri** [comment]
https://github.com/diffblue/cbmc/pull/6234 includes this fix.
